graphbolt/src/csc_sampling_graph.cc

```diff
@@ -781,10 +783,15 @@ inline int64_t LaborPick(
     std::iota(picked_data_ptr, picked_data_ptr + num_neighbors, offset);
     return num_neighbors;
   }
-  torch::Tensor heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
   // Assuming max_degree of a vertex is <= 4 billion.
-  auto heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
-      heap_tensor.data_ptr<int32_t>());
+  std::array<std::pair<float, uint32_t>, StackSize> heap;
+  auto heap_data = heap.data();
+  torch::Tensor heap_tensor;
+  if (fanout > StackSize) {
+    heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
+    heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
+        heap_tensor.data_ptr<int32_t>());
```

Comment:
`float`, `uint32_t`, `int32_t` are used in different places while we always assume all of them are in same size, namely `4` bytes. We may need a more secure way to violate it if someone else change one of them to different size such as `int64_t` some day.

Reply:
This is already secure. My only assumption is that a vertex doesn't have more than 4 billion neighbors. There is no dataset that I know where a vertex has more than 4 billion neighbors.

Reply:
I can change the code so that it allocates a tensor with enough bytes in guaranteed manner even if the types change. Let me do that.

Reply:
Just made a change.